codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -233,14 +234,21 @@ public ChannelFuture writePriority(ChannelHandlerContext ctx, int streamId, int
             }
 
             // Update the priority on this stream.
-            Http2Stream stream = connection.stream(streamId);
-            if (stream == null) {
-                stream = connection.local().createStream(streamId);
-            }
+            try {
+                Http2Stream stream = connection.stream(streamId);
+                if (stream == null) {
+                    stream = connection.local().createStream(streamId);
+                }
 
-            stream.setPriority(streamDependency, weight, exclusive);
-        } catch (Throwable e) {
-            return promise.setFailure(e);
+                // The set priority operation must be done before sending the frame. The parent may not yet exist
+                // and the priority tree may also be modified before sending.
+                stream.setPriority(streamDependency, weight, exclusive);
+            }  catch (ClosedStreamCreationException ignored) {
```

Comment:
Good point. Safe to assume `ClosedStreamCreationException` won't be thrown in the outer scope.